ConservationInternational/resilienceatlas
# feat: BE models and administration for static pages (About page)

I am slightly ahead of schedule with this feature. Based on investigation done within task https://vizzuality.atlassian.net/browse/RA2-168, this PR adds models and Administration which allows to create About page (Static pages).

Description of newly added models can be found at https://dbdiagram.io/d/64366ba28615191cfa8d4441 diagram. Basically there are 5 new tables:
 - one which is used as base for new static page
 - every static page can be composed of multiple sections
 - every section can be of different type --> paragraph, items or references. We use different tables to store different data for different section types.

Administration is using similar approach which was used for Journeys or Homepages. All entities can be created via one complex form --> individual sections are for example defined through `has_many` active admin method, etc. Individual sections are collapsible and based on selected section type, different part of form is visible. It is definitely easier for user to edit everything via one formular, but it is starting to feel quite complex already and we are really stretching what ActiveAdmin can do --> for even more complex structure (which hopefully will not come :D), It will be required to break it into multiple forms.

## Testing instructions

Login as admin and open Static Pages section. Try to create new static page which covers all important sections mentioned at https://www.resilienceatlas.org/about --> one paragraph, one person of Team, one Sponsor and for example one Reference. Verify that data can be saved and they are visible and editable at backoffice.

## Tracking

https://vizzuality.atlassian.net/browse/RA2-168
